Make react-datetime compatible with React 17.0.2

<!--
     **BEFORE PUBLISHING AN ISSUE**
  
  1) Make sure you are using the latest version of react-datetime. Execute
    npm ls react-datetime
  in your terminal to see what version you are using, and compare it to what verion is published to
  npm by looking at the package's npm page:
    https://www.npmjs.com/package/react-datetime
  
  2) Some of the most frequently asked questions have the answer for them in the documentation. 
  Be sure to read through the API carefully before publishing an issue: 
    https://github.com/arqex/react-datetime#api
  We also use the label *Documentation* for issues that have code examples in them which may be 
  useful for other developers, please have a look there as well: 
    https://github.com/arqex/react-datetime/issues?utf8=%E2%9C%93&q=label%3Adocumentation%20
  
  3) Discussions for version 3 of react-datetime has been initiated, and we have come up with some 
  feature and bug candidates. This means that if a feature or bug has been requested/reported many
  times, it's likely to be a candidate for version 3. Make sure your bug or feature is not included 
  in this list before publishing it.
  
  You can find all the candidates here in the *Candidates* column:
    https://github.com/arqex/react-datetime/projects/1
-->


### I'm Submitting a ...
<!-- Put an "x" in the box for the type of report that apply -->
```
[ ] Bug report
[ ] Feature request
[x] Support request
```

### Steps to Reproduce
<!-- Define the minimal steps to reproduce the bug, preferably in the format of a numbered list -->
I'm using npm version 7 and getting the following peerDependency error:
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: [myProject]
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   react@"^17.0.2" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.5.0" from react-datetime@3.0.4
npm ERR! node_modules/react-datetime
npm ERR!   react-datetime@"^3.0.4" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

### Expected Results
<!-- Describe what the expected results are -->

### Actual Results
<!-- Describe what the actual results are -->

### Minimal Reproduction of the Problem
<!--
  If the current behavior is a bug or you can illustrate your feature request better with an 
  example, please provide a *minimal demo* (if possible) of the problem via 
  http://codepen.io/ or similar. We recommend that you fork this example as a starting point, 
  it's using the *latest published* version of react-datetime: 
    http://codepen.io/simeg/pen/mEmQmP
-->

### Other Information (e.g. stacktraces, related issues, suggestions how to fix)
<!-- Any other information you think is valuable can be put here --> 


<!--
  Is there anything in this template you think is confusing, unclear, redundant or just simply bad?
  Please let us know either via creating an issue or creating a PR with changes to it.
-->

--- a/package.json
+++ b/package.json
@@ -38,7 +38,7 @@
   "license": "MIT",
   "peerDependencies": {
     "moment": "^2.16.0",
-    "react": "^16.5.0"
+    "react": "^16.5.0 || ^17.0.0"
   },
   "devDependencies": {
     "@babel/core": "7.7.4",

